### PR TITLE
Port launch command to support new Postgres and Redis launchers

### PR DESCRIPTION
**Pull request by jsierles**
- [x] Port launch command to new format
- [x] Make postgres launch/attach work from other commands
- [x] Fix deployment when a `fly.toml` exists

Note that this PR relies heavily on parameter structs being passed into functions that worked directly from CLI flags previously.

Also, a large set of new deployment-related flags were added. This may make the command more intimidating, but it also allows for various use cases requested by users who use `fly launch` in CI. If we don't want this, the new deploy command or `flags` would need fixing to ensure accessing non-existent flags doesn't panic.

### CODE REVIEW

**Review by rugwirobaker — COMMENTED** [review]
I encountered this panic on the first try:
```sh
Detected a Phoenix app
panic: flag accessed but not defined: dockerignore-from-gitignore

goroutine 1 [running]:
github.com/superfly/flyctl/internal/flag.GetBool({0x105d9d518?, 0x14000de30b0?}, {0x1056dfd02, 0x1b})
	/Users/codechef/workspace/fly.io/flyctl/internal/flag/context.go:71 +0x8c
github.com/superfly/flyctl/internal/command/launch.determineDockerIgnore({0x105d9d518, 0x14000de30b0}, {0x1400075b1a0, 0x1e})
	/Users/codechef/workspace/fly.io/flyctl/internal/command/launch/launch.go:692 +0xb0
github.com/superfly/flyctl/internal/command/launch.run({0x105d9d518, 0x14000de30b0})
	/Users/codechef/workspace/fly.io/flyctl/internal/command/launch/launch.go:232 +0xab0
github.com/superfly/flyctl/internal/command.newRunE.func1(0x14000dc4f00, {0x1068f9e60?, 0x0?, 0x0?})
	/Users/codechef/workspace/fly.io/flyctl/internal/command/command.go:110 +0x164
github.com/spf13/cobra.(*Command).execute(0x14000dc4f00, {0x1068f9e60, 0x0, 0x0})
	/Users/codechef/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000dc5180)
	/Users/codechef/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x354
github.com/spf13/cobra.(*Command).ExecuteContextC(...)
	/Users/codechef/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:911
github.com/superfly/flyctl/internal/cli.Run({0x105d9e190?, 0x14000ccf480?}, 0x14000754000, {0x140001b8010, 0x1, 0x1})
	/Users/codechef/workspace/fly.io/flyctl/internal/cli/cli.go:36 +0x1d0
main.run()
	/Users/codechef/workspace/fly.io/flyctl/main.go:45 +0x128
main.main()
	/Users/codechef/workspace/fly.io/flyctl/main.go:26 +0x24
```

File: internal/command/machine/launch.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func run(ctx context.Context) (err error) {
 		return
 	}
 
-	appConfig.WriteToDisk()
+	//	appConfig.WriteToDisk()
```
Comment (dangra) [inline internal/command/machine/launch.go:171]: what is going on here?

File: internal/command/launch/launch.go [hunk 2]
```diff
@@ -17,158 +17,158 @@ import (
...
-	return launchCmd
-}
-
-func runLaunch(cmdCtx *cmdctx.CmdContext) error {
-	ctx := cmdCtx.Command.Context()
-
-	dir := cmdCtx.Config.GetString("path")
+	cmd = command.New("launch", short, long, run, command.RequireSession, command.LoadAppConfigIfPresent)
+
+	cmd.Hidden = true
```
Comment (dangra) [inline internal/command/launch/launch.go:45]: why is it hidden?

**Review by dangra — COMMENTED** [review]
I tried `fly launch` for a small go app I use for testing. It failed on the last release step complaining about non existing app.

```
$ fly launch
Creating app in /Users/daniel/src/flocast-test
Scanning source code
Detected a Dockerfile app
? Choose an app name (leave blank to generate one): flocast-test
? Select Organization: Daniel Graña (personal)
? Choose a region for deployment: Miami, Florida (US) (mia)
Created app flocast-test in organization personal
? Would you like to set up a Postgresql database now? Yes
? Select configuration: Production - Highly available, 2x shared CPUs, 4GB RAM, 40GB disk
Creating postgres cluster flocast-test-db in organization personal
Creating app...
Setting secrets...
Provisioning 1 of 2 machines with image flyio/postgres:14.4
Waiting for machine to start...
Machine 568304ea73978e is created
Provisioning 2 of 2 machines with image flyio/postgres:14.4
Waiting for machine to start...
Machine 59185e16bd1983 is created
==> Monitoring health checks
  Waiting for 568304ea73978e to become healthy (started, 3/3)
  Waiting for 59185e16bd1983 to become healthy (started, 3/3)

Postgres cluster flocast-test-db created
  Username:    postgres
  Password:    ....
  Hostname:    flocast-test-db.internal
  Proxy port:  5432
  Postgres port:  5433
Save your credentials in a secure place -- you won't be able to see them again!

Connect to postgres
Any app within the Daniel Graña organization can connect to this Postgres using the following credentials:
For example: postgres://...@flocast-test-db.internal:5432


Now that you've set up postgres, here's what you need to understand: https://fly.io/docs/reference/postgres-whats-next/

Postgres cluster flocast-test-db is now attached to flocast-test
The following secret was added to flocast-test:
  DATABASE_URL=postgres://...@top2.nearest.of.flocast-test-db.internal:5432/flocast_test
Postgres cluster flocast-test-db is now attached to flocast-test
? Would you like to deploy now? Yes
==> Building image
Remote builder fly-builder-summer-glade-9382 ready
==> Creating build context
--> Creating build context done
==> Building image with Docker
--> docker host: 20.10.12 linux x86_64
[+] Building 1.5s (0/1)
[+] Building 31.1s (13/13) FINISHED
 (cut)
--> Building image done
==> Pushing image to fly
The push refers to repository [registry.fly.io/flocast-test]
31fea325302b: Pushed
994393dc58e7: Mounted from flocast
deployment-01GGAFJ9BDCR4EG2MGDZ1C3636: digest: sha256:30b97a7454fc2f0b4af42e824493317d2afde5c12a5eb2cfdebc9ef57c6f654c size: 738
--> Pushing image done
image: registry.fly.io/flocast-test:deployment-01GGAFJ9BDCR4EG2MGDZ1C3636
image size: 17 MB
==> Creating release
Error Could not find App

```

File: internal/command/machine/launch.go (hunk 1 above)
Comment (jsierles) [reply internal/command/machine/launch.go:171]: This is an aborted port of the launch command to the machines platform. This command is not visible to users, but we could remove it and revive later if desired.

File: internal/command/launch/launch.go (hunk 2 above)
Comment (jsierles) [reply internal/command/launch/launch.go:45]: Good catch. That was copypasted from the machines launch command.

**Comment by jsierles** [comment]
@dangra This should be good to go.

**Review by dangra — APPROVED** [review]

**Comment by dangra** [comment]
@jsierles hmm.. same thing on "fly pg create"

```
~$ fly pg create
? Choose an app name (leave blank to generate one):
? Select Organization: Daniel Graña (personal)
? Select region: Seattle, Washington (US) (sea)
? Select configuration: Development - Single node, 1x shared CPU, 256MB RAM, 1GB disk
Creating postgres cluster  in organization personal
Creating app...
Setting secrets...
Error Could not find App
```

**Comment by jsierles** [comment]
@dangra This actually happens in the current release. I'd guess most people are specifying an app name. If you don't, you see this failure. @rugwirobaker 

I'll look at fixing in this PR too.

**Comment by jsierles** [comment]
OK, that's fixed up.

**Pull request merged**